docs/yaml/functions/response_file.yaml

```diff
@@ -0,0 +1,67 @@
+name: response_file
+returns: rsp_file
+description: |
+  Write argument list to a response file, if the resulting length of the command
+  line is longer than a maximum value. A response file is a mechanism allowing a
+  program to receive more arguments than what would be allowed by the command
+  line. See [msvc compiler
+  documentation](https://learn.microsoft.com/en-us/cpp/build/reference/at-specify-a-compiler-response-file?view=msvc-170)
+  as an example.
+
+  This function is designed to be used with [[custom_target]], [[run_target]],
+  [[run_command]] or [[configure_file]].
+
+varargs:
+  name: arguments
+  type: str | file
+  description: The argument list.
+
+kwargs:
+  prefix:
+    type: str
+    default: "'@'"
+    description: |
+      The prefix to prepend to the response file path. Use an empty string if
+      you don't need a prefix.
+  
+  quote:
+    type: str | bool
+    default: true
+    description: |
+      Can be `'sh'` to use sh-like quoting, `'cmd'` for Windows cmd-like
+      quoting, or `'gcc'` for gcc-like quoting.
+
+      If `true`, uses `cmd` on Windows, or `sh` on other systems.
+
+      If `false`, does not quote the arguments.
+
+  max_args_length:
+    type: int
+    default: none
```

Comment:
Is there some better way to denote the platform dependent default

Reply:
What do you propose?

Reply:
`<platform dependent>` if refman will take it.